contracts/nns/contract.go

```diff
@@ -60,7 +60,7 @@ const (
 	maxTXTRecordLength = 255
 	// maxRecordID is the maximum value of record ID (the upper bound for the number
 	// of records with the same type).
-	maxRecordID = 255
+	maxRecordID = 16
```

Comment:
shouldnt this be `15` to `limit max number of records to 16`? and i dont think this needs to be separated from 51873d5a20715e6622b373fb2e4a49f634deb417

Reply:
> shouldnt this be 15

Correct, fixed!

>  i dont think this needs to be separated from 51873d5

Well, that's different from what @AnnaShaleva had originally, so it's either co-authored-by, or this. The last commit also updates binary contracts.

Reply:
reasonable then